### PR TITLE
Download log options more clear

### LINKED ISSUE #2565: Make download log data options more clear
You can select _shown interval_, _selected_ and _all data_ from the options. But _shown interval_ currently downloads whatever data has been fetched, which is limited to the current server's _maxDataNodes_ and _maxDataPoints_. I suggest these changes:

1. Rename "Download selected" to "Download selected rows"
2. Rename "Download shown interval" to "Download selected range" and change the functionality so that it orders the same job as when downloading all data, but with the additional startIndex and endIndex parameters.
3. Rename the title "Download" to "Download log data for 11 mnemonics" or something similar to make it clear that data will only be downloaded for the selected mnemonics.

Current options:
image

Index range and mnemonics selected for the log:
image

### PR DESCRIPTION
**Pull request by robertbasti**
[//]: # (Request Template Witsml Explorer)

[//]: # (Thank you for contributing to WITSML Explorer! Before submitting this PR, please fill in the following template.)

## Fixes

[//]: # (Write the GitHub issue number starting with #, or, for Equinor only, the Jira issue number starting with WE-)

This pull request fixes #2565 

## Description

[//]: # (Please include a written summary of the changes and which issue is fixed. List any dependencies that are required for this change._)

## Type of change

[//]: # (Mark any of the types of change that apply.)

* [ ] Bugfix
* [ ] New feature (non-breaking change which adds functionality)
* [x] Enhancement of existing functionality
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update

## Impacted Areas in Application

[//]: # (List general components of the application that this PR will affect)

* [x] Frontend
* [x] API
* [ ] WITSML
* [ ] Desktop
* [ ] Other (please describe)

## Checklist:

[//]: # (Please tick all the boxes or remove the ones that aren't needed and explain why)

*Communication*
* [ ] I have made corresponding changes to the documentation
* [ ] PR affects application security

*Code quality*
* [ ] I have self-reviewed my code
* [ ] No new warnings are generated

*Test coverage*
* [ ] New code is covered by passing tests

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: Src/WitsmlExplorer.Frontend/components/ContentViews/CurveValuesView.tsx [hunk 1]
```diff
@@ -219,13 +219,13 @@ export const CurveValuesView = (): React.ReactElement => {
...
+        exportSelectedRange();
         break;
       case DownloadOptions.SelectedIndexValues:
         exportSelectedDataPoints();
     }
-    downloadOptions = DownloadOptions.IntervalOfData;
+    downloadOptions = DownloadOptions.SelectedRange;
   };
 
   const exportSelectedIndexRange = useCallback(() => {
```
Comment (eliasbruvik) [inline Src/WitsmlExplorer.Frontend/components/ContentViews/CurveValuesView.tsx]: Remove this old function (only used in a dependency list, remove there also)

File: Src/WitsmlExplorer.Api/Jobs/DownloadAllLogDataJob.cs [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
 
 using WitsmlExplorer.Api.Models;
 
 namespace WitsmlExplorer.Api.Jobs;
 
 /// <summary>
-/// Job for downloading all log data.  
+/// Job for downloading all log data.
 /// </summary>
 public record DownloadAllLogDataJob : Job
```
Comment (eliasbruvik) [inline Src/WitsmlExplorer.Api/Jobs/DownloadAllLogDataJob.cs]: Now that it's not just for the whole log, can you remove _All_ to rename all occurrences of _DownloadAllLogData_ to _DownloadLogData_?

**Review by eliasbruvik — CHANGES_REQUESTED** [review]

File: Src/WitsmlExplorer.Api/Jobs/DownloadAllLogDataJob.cs (hunk 2 above)
Comment (robertbasti) [reply Src/WitsmlExplorer.Api/Jobs/DownloadAllLogDataJob.cs]: fixed

File: Src/WitsmlExplorer.Frontend/components/ContentViews/CurveValuesView.tsx (hunk 1 above)
Comment (robertbasti) [reply Src/WitsmlExplorer.Frontend/components/ContentViews/CurveValuesView.tsx]: fixed

**Review by eliasbruvik — APPROVED** [review]
*Reviewer checklist*
- [x] Read issue and PR
- [x] Pulled branch and manually tested
- [x] Verify that PR resolves issue
- [x] Reviewed the code

**Pull request merged**